.github/workflows/pr.yaml

```diff
@@ -0,0 +1,17 @@
+name: Testing
+on: [pull_request]
+
+# Declare default permissions as read only.
+permissions: read-all
```

Comment:
Is there a reason why we are granting `read-all` access? This lowers the security posture of the workflow

Reply:
scorecard suggests this in their docs
- https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions